src/js/carousel.js

```diff
@@ -1,21 +1,133 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 
-class Carousel extends React.Component {
+const styleArrow = {
+    zIndex: 1000,
+    backgroundColor: '#e3e3e3'
+};
+
+const styleSliderContainer = {
+    zIndex: 900, 
+    overflow: 'hidden'
+};
+
+class SliderContainer extends React.Component {
     constructor() {
         super();
         this.state = {
-            images: []
+            widthSlider: 0,
+            position: 0,
+            showingImages: 0
+        };
+
+        this.singleImageCointainerWidth = 0;
+
+        this.onClickArrow = this.onClickArrow.bind(this);
+        this.onClickImage = this.onClickImage.bind(this);
+        this.calculateDimensions = this.calculateDimensions.bind(this);
+    }
+
+    componentDidMount() {
+        window.addEventListener("resize", this.calculateDimensions);
+        this.calculateDimensions();
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("resize", this.calculateDimensions);
+    }
+
+    _getQuantityShowinImages(widthSliderContainer) {
```

Comment:
Si definis esto como una propiedad de la clase no necesitas hacer el bind arriba.
_getQuantityShowInImages = () => {
}

Reply:
si pero como todavía esta experimental eso lo hice así

Reply:
nunca hice el bind es una funcion privada que no usa el this

Reply:
Perdón, me confundí de función.